packages/build/src/utils/set.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const isPlainObj = require('is-plain-obj')
+
+// Set a property deeply using an array of `keys` which can be either strings
+// (object properties) or integers (array indices).
+// Adds default values when intermediary properties are undefined or have the
+// wrong type. Also extends arrays when they are too small for a given index.
+// Does not mutate.
+const setProp = function (parent, keys, value) {
```

Comment:
This function mostly does what `dot-prop` `set()` was doing. However, it uses an array of keys instead.
Surprisingly, I could not find any library which used an array of keys, supported arrays and did not directly mutate, so I had to implement one.